src/error.rs

```diff
@@ -67,13 +67,53 @@ impl Error {
     }
 
     pub(crate) fn response(mut self, response: Response) -> Self {
-        self.response = Some(Box::new(response));
+        self.response = Some(response);
         self
     }
-    pub(crate) fn kind(&self) -> ErrorKind {
+
+    /// The type of this error.
+    ///
+    /// ```
+    /// # ureq::is_test(true);
+    /// let err = ureq::get("http://httpbin.org/status/500")
+    ///     .call().unwrap_err();
+    /// assert_eq!(err.kind(), ureq::ErrorKind::HTTP);
+    /// ```
+    pub fn kind(&self) -> ErrorKind {
         self.kind
     }
 
+    /// For Errors of type HTTP (i.e. those that result from an HTTP status code),
+    /// return the status code of the response. For all other Errors, return 0.
+    ///
+    /// ```
+    /// # ureq::is_test(true);
+    /// let err = ureq::get("http://httpbin.org/status/500")
+    ///     .call().unwrap_err();
+    /// assert_eq!(err.kind(), ureq::ErrorKind::HTTP);
+    /// assert_eq!(err.status(), 500);
+    /// ```
+    pub fn status(&self) -> u16 {
```

Comment:
Maybe return an `Option` instead? Or an `Option<NonZeroU16>`?

Reply:
Oh interesting! I did not know about NonZeroU16. I went back and forth about whether to use Option instead of having 0 as a sentinel value. I think doing it this way simplifies calling code a lot, since callers are usually checking for specific values rather than trying to handle every possible outcome. For instance, with the sentinel value:

```
let res = ureq::get("http://example.com").call();
if let Err(e) = res {
  if e.status() > 500 {
    retry();
  }
}
```

vs (with Option):

```
let res = ureq::get("http://example.com").call();
if let Err(e) = res {
  if let Some(status) = e.status() {
    if status > 500 {
      retry();
    }
  }
}
```

Or, a caller could simplify that with:

```
let res = ureq::get("http://example.com").call();
if let Err(e) = res {
  if e.status().unwrap_or_default() > 500 {
    retry();
  }
}
```

But if we expect the caller to always call `unwrap_or_default`, we could save them the trouble!

Reply:
Ok, if the focus is for ergonomics I agree a sentinel value is nicer :)

Reply:
Yeah. I like sentinel.